src/pages/gen2/start/quickstart/nextjs-app-router-server-components/index.mdx

```diff
@@ -0,0 +1,346 @@
+
+import prerequisites from 'src/fragments/gen2/quickstart/prerequisites.mdx';
+
+<Fragments fragments={{ javascript: prerequisites, nextjs: prerequisites }} />
+
+import createProject from 'src/fragments/gen2/quickstart/create-nextjs-app-router-project.mdx';
+
+<Fragments fragments={{ javascript: createProject, nextjs: createProject }} />
+
+import buildABackend from 'src/fragments/gen2/quickstart/build-a-backend.mdx';
+
+<Fragments fragments={{ javascript: buildABackend, nextjs: buildABackend }} />
+
+
+## Build UI
+
+Let's add UI that connects to the backend data and auth resources.
+
+
+### Configure Amplify Client Side
+
+First, install the Amplify UI component library:
+
+```bash
+npm install @aws-amplify/ui-react
+```
+
+Next, create a `components` folder in the root of your project and the contents below to a file called `ConfigureAmplify.tsx`.
+
+```ts title="components/ConfigureAmplify.tsx"
+// components/ConfigureAmplify.tsx
+"use client";
+
+import { Amplify } from "aws-amplify";
+
+import config from "@/amplifyconfiguration.json";
+
+Amplify.configure(config, { ssr: true });
+
+export default function ConfigureAmplifyClientSide() {
+  return null;
+}
+```
+
+Update `app/layout.tsx` to import and render `<ConfigureAmplifyClientSide />`.  This client component will configure Amplify for client pages in our application.
+
+```ts title="app/layout.tsx"
+// app/layout.tsx
+import "@aws-amplify/ui-react/styles.css";
+import type { Metadata } from "next";
+import { Inter } from "next/font/google";
+import "./globals.css";
+
+import ConfigureAmplifyClientSide from "@/components/ConfigureAmplify";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata: Metadata = {
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <body className={inter.className}>
+        <ConfigureAmplifyClientSide />
+        {children}
+      </body>
+    </html>
+  );
+}
+```
+
+### Add a login page
+
+Create a new route under `app/login/page.tsx` to render the [Amplify Authenticator component](https://ui.docs.amplify.aws/react/connected-components/authenticator) when a user is not logged in or redirect to the root route.
+
+```ts title="app/login/page.tsx"
+// app/login/page.tsx
+
+"use client";
+
+import { redirect } from "next/navigation";
+
+import { AuthUser } from "aws-amplify/auth";
+import { withAuthenticator } from "@aws-amplify/ui-react";
+import { useEffect } from "react";
+
+function Login({ user }: { user: AuthUser }) {
+  useEffect(() => {
+    if (user) {
+      redirect("/");
+    }
+  }, [user]);
+  return null;
+}
+
+export default withAuthenticator(Login);
+```
+
+<Accordion title="Custom <Authenticator> example">
+
+Some applications require customizing the `<Authenticator>` component the following example shows how to add a custom Header to the `<Authenticator>`.
```

Comment:
@kevinold can we add a "such as/in order to" example here as to why?

Some applications require customizing the `<Authenticator>` component, such as/in order to.....  The following example shows how to add a custom Header to the `<Authenticator>`.